net-libs/libvncserver/libvncserver-0.9.12-r1.ebuild

```diff
@@ -0,0 +1,66 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit cmake-utils
+
+MY_PN="LibVNCServer"
+
+DESCRIPTION="library for creating vnc servers"
+HOMEPAGE="https://libvnc.github.io/"
+SRC_URI="https://github.com/LibVNC/${PN}/archive/${MY_PN}-${PV}.tar.gz"
+
+LICENSE="GPL-2"
+# No sub slot wanted (yet), see #578958
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~x86-fbsd ~amd64-linux ~x86-linux"
+IUSE="+24bpp gcrypt gnutls ipv6 +jpeg libressl lzo +png sasl ssl systemd +threads +zlib"
+REQUIRED_USE="!gnutls? ( ssl? ( threads ) ) png? ( zlib )"
```

Comment:
this looks strange; upstream `CMakeLists.txt` says this is about cryptography, and it would fallback to builtin functions if `!gnutls` and `!openssl`. So USE `ssl` gets it right mechanically, in a way, but does not express it properly. @SoapGentoo what do you think?
methinks `/IUSE/s/ssl/+system-crypto/`, then a cascade of `system-crypto? ( gnutls? (...) !gnutls? ( libressl... )`, at least logically.

Reply:
AFAIS, builtin crypto functions are just stubs. See `libvncclient/tls_none.c` and `libvncserver/rfbssl_none.c`

Reply:
I probably should add comment before REQUIRED_USE with bugs links, because these USE restrictions are not forced by build system.
https://bugs.gentoo.org/550916
https://bugs.gentoo.org/435326

Reply:
> AFAIS, builtin crypto functions are just stubs. See `libvncclient/tls_none.c` and `libvncserver/rfbssl_none.c`

I wasn't totally right. There are indeed internal crypto functions for md5 and sha1 hashing. But openssl and gnutls are also used for TLS, not just crypto.

Reply:
So after fixing gnutls dependencies:
- `gcrypt` - use libgcrypt-based crypto + support for additional authentication methods
- `openssl` - use openssl-based crypto (but libgcrypt is preferred if enabled) + TLS via openssl
- `gnutls` - TLS via gnutls
- `-openssl -gcrypt` - use internal crypto
- `-openssl -gnutls` - no TLS